### PR TITLE
[tts][TCSACR-477] Add guide for Repeat()

### PR DESCRIPTION
**Pull request by stom-hwang**
### Change Description ###

Add description for new API `Repeat()`

### API Changes ###

 - ACR: TCSACR-477

### CODE REVIEW

File: docs/application/dotnet/guides/uix/tts.md [hunk 1]
```diff
@@ -593,6 +593,38 @@ To start, pause, and stop the playback:
     ```
 
 
+-   To repeat the last playback, use the `Repeat()` method.
+
+    The `Repeat()` method returns the information about the text which is last played, and the TTS state is changed to `Playing`.
```
[the remark this answers is not in the document]
Comment (stom-hwang) [reply docs/application/dotnet/guides/uix/tts.md:598]: 네 정확히는 RepeatedText라고 하는 클래스이구요, 해당 클래스에는 마지막으로 재생했던 text string과 해당 text에 대한 utterance id가 포함되어 있습니다. RepeatedText 클래스에 대한 설명도 추가하는것이 더 좋을까요?

**Review by asm-foysal — COMMENTED** [review]
Line 6: In the last sentence,  "This daemon and your application communicate as the server and the client", to specify the server and the client, we can add respectively at the end of the sentence.
Suggestion: "This daemon and your application communicate as the server and the client, respectively". Here the comma before respectively helps to signify the syntactic function (the order or arrangement of separate elements). 

Line 23: Please use sentence-style capitalization for second headings. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#capitalization
Suggestion: Basic TTS processes

Line 45: Please use sentence-style capitalization for second headings. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#capitalization
Suggestion: TTS parameters

Line 58: Please use sentence-style capitalization for second headings. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#capitalization
Suggestion: TTS information retrieval

Line 66: "The user can change the default voice in the device settings, by modifying the display language or the TTS default language status"
Suggestion: "The user can change the default voice in the device settings by modifying the display language or the TTS default language status",  the comma can be eliminated here. 

Line 123: Requesting to use sentence-style capitalization & eliminate gerund form. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#verb-form
Correction: Register event handlers

Line 280: Requesting to use sentence-style capitalization & eliminate gerund form. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#verb-form
Correction: Get information

Line 286: "The TTS state is changed by various TTS methods, and it is applied as a precondition for each method", 
Suggestion: We should add a colon at the end of the sentence since a code section starts from the next line. 

Line 322: "TTS synthesizes the text using the default voice, if you do not set the language and the voice type as parameters of the `AddText()` method of the `Tizen.Uix.Tts.TtsClient` class."
Suggestion: We should add a colon at the end of the sentence since a code section starts from the next line. 

Line 341: Requesting to use sentence-style capitalization & eliminate gerund form. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#verb-form
Suggestion: Get and set the mode

Line 377: Requesting to use sentence-style capitalization & eliminate gerund form.
Suggestion: Connect and disconnect TTS

Line 383: "The daemon synthesizes the text with the engine and plays the resulting sound data. The method is asynchronous and the TTS state changes to `Ready`."
Suggestion: We should add a colon at the end of the sentence since a code section starts from the next line.

Line 419: Requesting to use sentence-style capitalization & eliminate gerund form. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#verb-form
Suggestion: Set and get TTS engine options

Line 492: "You can add text to TTS at any point after the `Prepare()` method changes the TTS state to `Ready`."
Suggestion: We should add a colon at the end of the sentence since a code section starts from the next line.

Line 530: Requesting to use sentence-style capitalization & eliminate gerund form. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#verb-form
Suggestion: Control playback

Line 538: The last sentence, "The TTS state need not change to `Ready` before using the `Stop()` method", is not clear to me. It would be helpful if you can explain the gist of the sentence. I might be able to deliver another perspective in making it easier to understand. 

Line 542: The Note section sounds a little confusing to me. Maybe it can be rephrased using proper punctuation. If you can explain the sentence further, then I can try to offer a suggestion to restructure the note.

Line 561: "The TTS state is changed to `Paused`. To resume playback, use the `Play()` method"
Suggestion: We should add a colon at the end of the sentence since a code section starts from the next line.

Line 579: "All the texts in the queue are removed, and the TTS state is changed to `Ready`"
Suggestion: We should add a colon at the end of the sentence since a code section starts from the next line.

File: docs/application/dotnet/guides/uix/tts.md [hunk 2]
```diff
@@ -593,6 +593,39 @@ To start, pause, and stop the playback:
     ```
 
 
+-   To repeat the last playback, use the `Repeat()` method.
+
+    The `Repeat()` method returns the information about the text is played lastly, and the TTS state is changed to `Playing`.
+
+
+    If there is no text is played before,  an exception occurs. To handle the exception, contain a block of code as follow:
```
[the remark this answers is not in the document]
Comment (stom-hwang) [reply docs/application/dotnet/guides/uix/tts.md:601]: As the comment that I leave previously, `Repeat()` uses the lastly added text, so I am going to fix like this.
```suggestion
    If no text has been added before, an exception occurs. To handle the exception, contain a block of code as follow: 
```

File: docs/application/dotnet/guides/uix/tts.md [hunk 3]
```diff
@@ -593,6 +593,39 @@ To start, pause, and stop the playback:
...
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+
 ## Related Information
```
[the remark this answers is not in the document]
Comment (stom-hwang) [reply docs/application/dotnet/guides/uix/tts.md:629]: Thank you for checking.
I will fix it soon.

File: docs/application/dotnet/guides/uix/tts.md [hunk 4]
```diff
@@ -593,6 +593,39 @@ To start, pause, and stop the playback:
...
 * Dependencies
```
[the remark this answers is not in the document]
Comment (stom-hwang) [reply docs/application/dotnet/guides/uix/tts.md:630]: This seems to be error by copying.
I will fix it soon.
Thank you for checking.

**Review by stom-hwang — COMMENTED** [review]
Thank you for detailed review.
I will fix the guide soon.
And, I leave the comment about `Repeat()`.
Would you please check the comment and share your opinion?

**Comment by stom-hwang** [comment]
@asm-foysal
Hello, I have fixed the guide to follow your review and left comments.
Please check this PR.

**Review by asm-foysal — COMMENTED** [review]
@stom-hwang Thank you for incorporating all the previously requested changes. 
We went through the entire file again to make sure we haven't missed anything, hence the delay with this feedback.
We came across some issues during this review which has been detailed below:

1. Line 481 – “Adding Text”
**_Suggestion_** - “Add text”
Reference – It is best not to use gerund form to start heading,
https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#capitalization
For second headings, please use sentence-style capitalization
https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#capitalization

2. Line 538 – “If there is no text in the queue, TTS waits in the `Playing` state for text to be added. In that case, when you add text, TTS starts synthesizing and playing it immediately. The TTS state need not change to `Ready` before using the `Stop()` method.”
**_Suggestion_** - “If there is no text in the queue, TTS stays in the `Playing` state for text to be added. In that case, when text is added, TTS starts synthesizing and playing it immediately. There is no need to change the state to ‘Ready’ by using the ‘Stop()’ method even when there is no text in the queue. It will continue to stay in the playing state.”

3. Line 542 – “If the TTS state changed event handler is invoked in the `Playing` state without a TTS method call, prepare the TTS state. The state can change if other applications request TTS play, the audio session requests TTS pause, or the TTS engine changes.”
**_Suggestion_** – If the TTS state changed event handler is invoked, state change will occur without any controlling methods being called. The state change takes place only when other applications request TTS play, the audio session requests TTS pause, or the TTS engine changes.

Line – 85, 119, 345, 349, 445, 487, 540
**_Suggestion_** – We noticed that the “**Note**” in this file follows the previous format, please change them according to the new alert format.
Reference – For alerts, please use the alerts formatting as per the guide.
https://github.com/Samsung/tizen-docs/blob/master/styleguide/custom-style.md#alerts-note-tip-important-caution-warning

**Comment by stom-hwang** [comment]
@asm-foysal 
Thank you for review. I have few questions.

1. Line 542 - How about this sentences? Actually, state changed event handler can be also invoked by calling controlling methods such as `Play()`, `Stop()`, `Pause()`, and so on. However, it seems that your seggestion says that state change occurs only when other applications behavior.
***Suggestion*** - If the TTS state changed event handler is invoked, state change can occur without any controlling methods being called. In this case, the state change takes place only when other aplications request TTS play, the audio session requests TTS pause, or the TTS engine changes.

2. Line 596 - I would like to change the guide of `Repeat()` as I commented on the line. so, would you please check this sentences?
***Suggestion*** - To repeat the last added text, use the `Repeat()` method.
The `Repeat()` method returns the information about the text is added lastly, and the TTS state is changed to `Playing`.
If no text has been added before, an exception occurs. To handle the exception, contain a block of code as follow:

**Review by asm-foysal — COMMENTED** [review]
Hi @stom-hwang! Sorry I didn't address these in my earlier review. 
Unfortunately, I'm not being able to comment on the conversation, hence I have to post another review. 

1.	Line 542 – Suggestion made by you– “If the TTS state changed event handler is invoked, a state change can occur without any controlling methods being called. In this case, the state change takes place only when other applications request TTS play, the audio session requests TTS pause, or the TTS engine changes.”
> You are right. Also, the change that you made clarifies the confusion we had, please go ahead and incorporate it.

2.	Line 598 – “The `Repeat()` method returns the information about the text is played lastly, and the TTS state is changed to `Playing`.”
Suggestion – “The `Repeat()` method returns the information about the last added text and the TTS state is changed to `Playing`.”
>The word “played” has been changed to “added” and the part "...text is played lastly" has also been addressed. I think this delivers the message properly, what do you think?
If the changes make sense to you, please go ahead and incorporate them, or else we can have a further discussion regarding this.

**Comment by stom-hwang** [comment]
@asm-foysal 
Thank you for response. 😊
Your suggestion looks nice to me. I will fix and upload new commit soon.

File: docs/application/dotnet/guides/uix/tts.md [hunk 5]
```diff
@@ -478,18 +478,18 @@ To set and get TTS engine options:
     ```
 
 <a name="text"></a>
-## Adding Text
+## Add Text
```
Comment (asm-foysal) [inline docs/application/dotnet/guides/uix/tts.md:481]: I think the second heading capitalization issue was not addressed.
```suggestion
## Add text
```

**Review by asm-foysal — COMMENTED** [review]
@stom-hwang one small issue, please address this.

@mijin85cho this PR is almost ready for merging.

File: docs/application/dotnet/guides/uix/tts.md (hunk 5 above)
Comment (stom-hwang) [reply docs/application/dotnet/guides/uix/tts.md]: Thank you for checking.
I also missed it.
I will fix it.

**Review by asm-foysal — APPROVED** [review]
@stom-hwang thank you for incorporating all the changes.

@mijin85cho I'm approving this PR for merging.

**Pull request merged**